File: app/assets/javascripts/Composition.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const CompositionSchema = DataElementSchema({
  identifier: Any,
  status: Any,
  type: Any,
  category: Any,
  subject: Any,
  encounter: Any,
  date: Any,
  author: Any,
  title: Any,
  confidentiality: Any,
  attester: Any,
  custodian: Any,
  relatesTo: Any,
  event: Any,
  section: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'Composition' },

});

module.exports.CompositionSchema = CompositionSchema;
module.exports.Composition = mongoose.model('Composition', CompositionSchema);

File: app/models/qdm/composition.rb
module QDM
  # app/models/qdm/composition.rb
  class Composition < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :status
    field :type
    field :category
    field :subject
    field :encounter
    field :date
    field :author
    field :title
    field :confidentiality
    field :attester
    field :custodian
    field :relatesTo
    field :event
    field :section
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
